panda/src/device/ioKitInputDeviceManager.cxx

```diff
@@ -78,16 +79,33 @@ IOKitInputDeviceManager::
  */
 void IOKitInputDeviceManager::
 on_match_device(void *ctx, IOReturn result, void *sender, IOHIDDeviceRef device) {
-  InputDeviceManager *mgr = (InputDeviceManager *)ctx;
+  IOKitInputDeviceManager *mgr = (IOKitInputDeviceManager *)ctx;
   nassertv(mgr != nullptr);
   nassertv(device);
 
-  PT(InputDevice) input_device = new IOKitInputDevice(device);
+  PT(IOKitInputDevice) input_device = new IOKitInputDevice(device);
   if (device_cat.is_debug()) {
     device_cat.debug()
       << "Discovered input device " << *input_device << "\n";
   }
   mgr->add_device(input_device);
+  mgr->_devices_by_hidref[device] = input_device;
 }
 
+/**
+ * Called by IOKit when an input device has disappeared.
+ */
+void IOKitInputDeviceManager::
+on_remove_device(void *ctx, IOReturn result, void *sender, IOHIDDeviceRef device) {
+  IOKitInputDeviceManager *mgr = (IOKitInputDeviceManager *)ctx;
+  nassertv(mgr != nullptr);
+  nassertv(device);
+
+  nassertv(mgr->_devices_by_hidref.find(device) != mgr->_devices_by_hidref.end());
+  PT(IOKitInputDevice) input_device = mgr->_devices_by_hidref[device];
```

Comment:
We don't need to increase the reference count here (we've already asserted that there exists a reference to it, and will be removing it at the end of this method so the PT doesn't serve to extend the lifetime of any reference).  So a regular pointer will suffice.

This is pretty insignificant, but I figured I might mention it while I'm at it: we're also unnecessarily accessing the map twice here.  It'd be more efficient to replace the two lines above with this:
```
auto it = mgr->_devices_by_hidref.find(device);
nassertv(it != mgr->_devices_by_hidref.end());
IOKitInputDevice *input_device = *it;
```